### ISSUE TITLE
0.17.0  Cannot automatically detect app

### ISSUE THREAD

**Issue by jmoody** [calabash/calabash-ios#959]
@arulami reported.

```ruby
TypeError: no implicit conversion of nil into String
from /Users/xxxx8/.rvm/gems/ruby-2.1.5/gems/calabash-cucumber-0.17.0/lib/calabash-cucumber/launch/simulator_launcher.rb:255:in join' from /Users/xxxx8/.rvm/gems/ruby-2.1.5/gems/calabash-cucumber-0.17.0/lib/calabash-cucumber/launch/simulator_launcher.rb:255:inapp_bundle_or_raise'
from /Users/xxxx8/.rvm/gems/ruby-2.1.5/gems/calabash-cucumber-0.17.0/lib/calabash-cucumber/launcher.rb:541:in relaunch' from /Users/xxxxx/.rvm/gems/ruby-2.1.5/gems/calabash-cucumber-0.17.0/lib/calabash-cucumber/core.rb:943:instart_test_server_in_background'
from (irb):1
from /Users/a042468/.rvm/rubies/ruby-2.1.5/bin/irb:11:in `
```

* Was working in 0.16.4. 
* Works if `APP` is set.

This does not work:

```
$ DEBUG=1 APP=/path/to/My.app calabash-ios console
> start_test_server_in_background
```

**Comment by jmoody** [comment]
@arulami, in your original comment you say that this does not work:

```
$ DEBUG=1 APP_BUNDLE_PATH=/path/to/My.app calabash-ios console
> start_test_server_in_background
```

Does it fail in the same way?

**Comment by arulami** [comment]
While running below command i have got an error was below:

DEBUG=1 APP_BUNDLE_PATH="~/Library/Developer/Xcode/DerivedData/xfgxfgxfgxfgfxg/Build/Products/Debug-iphonesimulator/Pricxxxx.app" calabash-ios console

Also I have  created cal-target for iPad 2 (9.2) but it shows in error as 

RuntimeError: Could not find simulator with name or UDID that matches: 'iPhone 6s (9.1)'


INFO: Using uia strategy: 'host'
DEBUG: Searching for run-loop results with glob: /Users/a042468/.run-loop/results/*
DEBUG: Found 6 previous run-loop results
DEBUG: Will delete 1 previous run-loop results
DEBUG: Deleted 1 previous results in 0.004754 seconds
DEBUG: Searching for instruments caches with glob: /Library/Caches/com.apple.dt.instruments/xrtmp__*
DEBUG: Found 5 instruments caches
DEBUG: Will delete 0 instruments caches
DEBUG: Deleted 0 instruments caches in 0.000103 seconds
RuntimeError: Could not find simulator with name or UDID that matches: 'iPhone 6s (9.1)'
	from /Users/a042fgg468/.rvm/gems/ruby-2.1.5/gems/run_loop-2.0.1/lib/run_loop/core.rb:178:in `prepare_simulator'
	from /Users/a04ewe2468/.rvm/gems/ruby-2.1.5/gems/run_loop-2.0.1/lib/run_loop/core.rb:297:in `run_with_options'
	from /Users/a0ewe42468/.rvm/gems/ruby-2.1.5/gems/run_loop-2.0.1/lib/run_loop.rb:100:in `run'
	from /Users/a04ewe2468/.rvm/gems/ruby-2.1.5/gems/calabash-cucumber-0.17.0/lib/calabash-cucumber/launcher.rb:714:in `block in new_run_loop'
	from /Users/a042ewew468/.rvm/gems/ruby-2.1.5/gems/calabash-cucumber-0.17.0/lib/calabash-cucumber/launcher.rb:712:in `times'
	from /Users/a042ewew468/.rvm/gems/ruby-2.1.5/gems/calabash-cucumber-0.17.0/lib/calabash-cucumber/launcher.rb:712:in `new_run_loop'
	from /Users/a042ewe468/.rvm/gems/ruby-2.1.5/gems/calabash-cucumber-0.17.0/lib/calabash-cucumber/launcher.rb:580:in `relaunch'
	from /Users/a042ewe468/.rvm/gems/ruby-2.1.5/gems/calabash-cucumber-0.17.0/lib/calabash-cucumber/core.rb:943:in `start_test_server_in_background'
	from (irb):1
	from /Users/a042eew468/.rvm/rubies/ruby-2.1.5/bin/irb:11:in `<main>'

**Comment by jmoody** [comment]
Please use GitHub formatting in your comments and issues.

**Comment by jmoody** [comment]
```
RuntimeError: Could not find simulator with name or UDID that matches: 'iPhone 6s (9.1)'
```

This is a completely different error and it is a bug that will be solved by run-loop 2.0.2 which is in progress.

You can clear this bug by:

* Installing more Simulators from Xcode and restarting your computer.
* setting `DEVICE_TARGET="iPhone 6s (9.2)"`

**Comment by arulami** [comment]
RuntimeError: Could not find simulator with name or UDID that matches: 'iPhone 6s (9.1)'

Above error is solved by running below comment:
DEBUG=1 DEVICE_TARGET='iPad 2 (9.2)' APP_BUNDLE_PATH="~/Library/Developer/Xcode/DerivedData/PriceMatch-gtdnenfnhcypqyeuehsewfpuxlvb/Build/Products/Debug-iphonesimulator/PriceMatch.app" calabash-ios console

Main Issue: 0.17.0 Cannot automatically detect app #959 has to be fixed.
